### PR TITLE
Fix test results marshalling into different files

### PR DESCRIPTION
**Pull request by iamsubratp**
Currently, the results are concatenated and dumped into a single file
which is parsed and results are shown on testgrid. With multiple testsuites
in place, the xml generated does not have a root element resulting in failure
in parsing the xml.

### CODE REVIEW

File: osconfig_tests/main.go [hunk 1]
```diff
@@ -85,23 +81,25 @@ func main() {
 		close(tests)
 	}()
 
-	var testSuites []*junitxml.TestSuite
 	for ret := range tests {
-		testSuites = append(testSuites, ret)
-	}
+		testSuiteOutPath := fmt.Sprintf("%s-%s", ret.Name, *outPath)
```
Comment (adjackura) [inline osconfig_tests/main.go]: outPath is supposed to be a path, maybe we need to change the flag to outdir? Then this would be filepath.Join

Comment (iamsubratp) [reply osconfig_tests/main.go]: yeah, changed the outdir.

File: osconfig_tests/main.go [hunk 2]
```diff
@@ -85,23 +81,25 @@ func main() {
 		close(tests)
 	}()
 
-	var testSuites []*junitxml.TestSuite
 	for ret := range tests {
-		testSuites = append(testSuites, ret)
-	}
+		testSuiteOutPath := filepath.Join(*outDir, fmt.Sprintf("%s.xml", ret.Name))
```
Comment (adjackura) [inline osconfig_tests/main.go]: For clarity:
fmt.Sprintf("%s_junit.xml", ret.Name)

File: osconfig_tests/main.go (hunk 1 above)
Comment (adjackura) [reply osconfig_tests/main.go]: Also update config.yaml in test-infra/prow with this new flag

File: osconfig_tests/main.go (hunk 2 above)
Comment (iamsubratp) [reply osconfig_tests/main.go]: done

File: osconfig_tests/main.go (hunk 1 above)
Comment (iamsubratp) [reply osconfig_tests/main.go]: done

**Pull request merged**